playbooks/create_vm.yaml

```diff
@@ -16,32 +20,42 @@
         map('extract', hostvars, 'vm_host') |
         unique
       }}
-    _missing_hosts: |-
-      {{ _vm_hosts | difference(ansible_play_hosts_all) }}
+    _invalid_hosts: |-
+      {{ ansible_play_hosts_all | difference(groups['libvirt_guests']) }}
   tasks:
-    - name: Check VM hosts are included in execution
+    - name: Check target hosts are in libvirt_guests group
       ansible.builtin.assert:
         that:
-          - not _missing_hosts
+          - not _invalid_hosts
         quiet: true
         fail_msg: |-
-          The VM hosts should be included in the play: {{ _missing_hosts }}.
+          The VM hosts should be included in the libvirt_guests group:
+          {{ _invalid_hosts }}.
       run_once: true
 
     - name: Create transient VM hosts group
       ansible.builtin.add_host:
         name: '{{ item }}'
         group: vm_hosts
-        # Save playbook targets, not limited by play `hosts` param.
-        _limit_hosts: '{{ ansible_play_hosts_all }}'
+        # Save playbook targets as a list of hosts
+        _target_hosts: '{{ ansible_play_hosts_all }}'
       loop: '{{ _vm_hosts }}'
       check_mode: false
       changed_when: false
 
 - name: Prepare host
   hosts: vm_hosts
+  gather_facts: true
   become: true
   tasks:
+    - name: Install dependencies
+      ansible.builtin.package:
+        state: present
+        name:
+          - guestfs-tools
+          - python3-libvirt
+          - virtinst
```

Comment:
Should `libvirt-daemon` included here? (I noticed it was deleted from libvirt_hosts.yaml)

Reply:
No, I removed it because the `libvirt_host` role takes care of installing it